### DISCUSSION TITLE
Panel alignment with/without left Y-axis

### DISCUSSION THREAD

**Discussion by BitProcessor** [grafana/grafana#38261] [Dashboards, Panels & Visualizations]
It would be nice if panels without Left Y Axis would align with panels that do have the Y axis. 
That would make it a lot nicer visually and also more easy to compare graphs time-wise. 

As you can see in the screenshot:

- left and right side are aligned (black arrows)
- rightmost timeline is aligned between both graphs (blue line)
- leftmost (and intermediate ones) timeline is NOT aligned between both graphs (red line)

![alignment](https://user-images.githubusercontent.com/50638541/114831293-829b3b80-9dcd-11eb-8488-721da9c8efc0.png)

**Comment by leeoniya** [comment]
this might be pretty tough to do with axis auto-sizing. a much simpler path would be to add support for insetting the axis labels:

https://jsfiddle.net/580jdL7n/

![image](https://user-images.githubusercontent.com/43234/115072584-6676c780-9ebd-11eb-859b-4c3e738a4eb9.png)

**Comment by BitProcessor** [comment]
Could it be an option that 'hiding' the Axis simply sets the text to the backgroundcolor? Wouldn't that keep the alignment the same with a panel that has a 'visible' Y-Axis ?

The option presented above doesn't really look nice either if you ask me. Personal opinion of course.

**Comment by leeoniya** [comment]
> The option presented above doesn't really look nice either if you ask me. Personal opinion of course.

i agree. however, there's _a lot_ to figure out here because anything we do has to support a vast number of dashboard configs and arrangements, not just one with two stacked time series graphs.

1. aligning y axis widths is only useful if the graphs are vertically stacked. otherwise it would look weird, with empty unused space. this means trying to understand dashboard layouts composed of maybe 15 differently sized and positioned panels of all kinds?
2. the width of each y axis is determined by the tick text that would be rendered there, which in turn is determined by that y data range in view for that unit. there is currently no cross-coordination between graphs to correlate chart geometry and making something that is automatic and also correct seems very difficult.
3. something like this willl almost certainly impact performance would require multiple re-renders. imagine 20 panels that have queries that come in 200ms apart, for each response you'd need to recalculate the maximum axis width and rerender all "linked" charts on each query response.
4. how do you choose which panels should be aligned/linked and which shouldn't?
5. there can be an unlimited number of y axes in each panel, on different sides.
6. ... and 100 more considerations.

in my view, any kind of automatic axis width sync will be extemely fragile, imperfect, slow, and complex.

we currently have an option to set an exact axis width, which is extremely simple, intuitive and easily solves your original request. you do give up automatic width adjustment based on axis range and units, but i think the trade-off is quite reasonable.

i'd be interested to hear your approach to addressing each of the issues i've listed above.

**Comment by Strunck** [comment]
The y-axis width is rendered by the tick text.  If the user could format the tick text, oneself could ensure the correct width. 
Maybe something like this:
Format Tick text: 00.00 ### 
would render to 
01.23 bar 
21.33  °C

It is up to the user to ensure the formating.

**Comment by leeoniya** [comment]
how would this be different than defining a static axis width? (which is already possible)

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
